### ISSUE TITLE
C2D V2 support - Add fileObject types

### ISSUE THREAD

**Issue by paulo-ocean** [oceanprotocol/ocean.js#1867]
extend 
```
export interface ComputeAsset {
  **fileObject: BaseFileObject**
 documentId: string
  serviceId: string
  transferTxId?: string
  userdata?: { [key: string]: any }
...

```
By adding **fileObject**

For reference check this Ocean Node PR: [https://github.com/oceanprotocol/ocean-node/pull/705](https://github.com/oceanprotocol/ocean-node/pull/705)
...
export enum EncryptMethod {
  AES = 'AES',
  ECIES = 'ECIES'
}

export interface BaseFileObject {
  type: string
  encryptedBy?: string
  encryptMethod?: EncryptMethod
}

Also add support for start C2D on free compute Environments ( new route: `freeCompute` )

**Comment by jamiehewitt15** [comment]
Is this closed by #1870 ?

**Comment by paulo-ocean** [comment]
> Is this closed by #1870 ?

yes, it is
